pkg/zsio/ndjson/ndjson_test.go

```diff
@@ -7,11 +7,43 @@ import (
 	"testing"
 
 	"github.com/mccanne/zq/pkg/zsio/ndjson"
+	zsonio "github.com/mccanne/zq/pkg/zsio/zson"
 	"github.com/mccanne/zq/pkg/zson"
 	"github.com/mccanne/zq/pkg/zson/resolver"
 	"github.com/stretchr/testify/require"
 )
 
+func TestNDJSONWriter(t *testing.T) {
+	type testcase struct {
+		name, input, output string
+	}
+	cases := []testcase{
+		{
+			name: "null containers",
+			input: `#0:record[dns:vector[string],uri:vector[string],email:set[string],ip:vector[addr]]
+0:[[google.com;]-;-;-;]
+`,
+			output: `{"dns":["google.com"],"email":null,"ip":null,"uri":null}`,
+		},
+		{
+			name: "nested nulls",
+			input: `#0:record[san:record[dns:vector[string],uri:vector[string],email:set[string],ip:vector[addr]]]
+0:[[[google.com;]-;-;-;]]
+`,
+			output: `{"san":{"dns":["google.com"],"email":null,"ip":null,"uri":null}}`,
+		},
```

Comment:
Maybe include a test handles an empty vector or set?

Reply:
Yes I realized after putting up this PR that there's a bug there too... will add and fix.

Reply:
added tests for empties, will merge.

(turns out there was no bug there... my test was  wrong. )